api/handlers.go

```diff
@@ -330,6 +330,42 @@ func getVolume(c *context, w http.ResponseWriter, r *http.Request) {
 	httpError(w, fmt.Sprintf("No such volume: %s", name), http.StatusNotFound)
 }
 
+func getUsedVolumes(containers []*cluster.Container) map[string]bool {
+	usedVolumes := make(map[string]bool)
+	for _, container := range containers {
+		for _, mount := range container.Mounts {
+			if mount.Type == "volume" {
+				usedVolumes[container.Engine.Name+"/"+mount.Name] = true
+			}
+		}
+	}
+	return usedVolumes
+}
+
+func getFilteredVolumes(containers []*cluster.Container, volumes []*apitypes.Volume, dangling bool) []*apitypes.Volume {
+	usedVolumes := getUsedVolumes(containers)
+	var filteredVolumes []*apitypes.Volume
+	// volume name allowed char [a-zA-Z0-9][a-zA-Z0-9_.-]
+	// splitting on / is safe to keep the original output intact
+	stripEngine := func(vol *apitypes.Volume) *apitypes.Volume {
+		//engine name is only appended to driver "local" volumes (see: getVolumes)
+		//this naming format is temporarily extended to all volumes to provide uniqueness
+		// while identifying dangling volumes. Restoring the original naming convention back.
+		if vol.Driver != "local" {
+			vol.Name = strings.Split(vol.Name, "/")[1]
```

Comment:
Would it make sense to check that `strings.Split(vol.Name, "/")` here has `len() > 1` so that we don't panic if there's a problem here?

Reply:
Possibly SplitN as well (I think with custom drivers, a slash may be allowed in names)

Reply:
here https://github.com/dani-docker/swarm/blob/117d215985d6d06fb66115dd781e32253fe3aa79/api/handlers.go#L426 guarantees that "/" is appended when dangling filter is in use. 
But if you think this helper method will be re-used and to be on the safe, I will make the change.

Reply:
@thaJeztah 
you are correct! when I looked at the code, I only checked local volume, as you stated, custom drivers  so allow slash

Reply:
changes done for both concerns